fe/fe-core/src/main/java/com/starrocks/persist/GlobalVarPersistInfo.java

```diff
@@ -78,73 +78,89 @@ public String getPersistJsonString() {
         return persistJsonString;
     }
 
+    public List<String> getVarNames() {
+        return varNames;
+    }
+
     @Override
     public void write(DataOutput out) throws IOException {
         try {
-            JSONObject root = new JSONObject();
-            for (String varName : varNames) {
-                // find attr in defaultSessionVariable or GlobalVariables
-                Object varInstance = null;
-                Field theField = null;
-                boolean found = false;
-                // 1. first find in defaultSessionVariable
-                for (Field field : SessionVariable.class.getDeclaredFields()) {
-                    VariableMgr.VarAttr attr = field.getAnnotation(VariableMgr.VarAttr.class);
-                    if (attr == null) {
-                        continue;
-                    }
-                    if (attr.name().equalsIgnoreCase(varName)) {
-                        varInstance = this.defaultSessionVariable;
-                        theField = field;
-                        found = true;
-                        break;
-                    }
+            JSONObject root;
+            // Handle refresh connections command (empty varNames)
+            if (varNames == null || varNames.isEmpty()) {
```

Comment:
Avoid mixing special fields with normal logic, as this will lead to maintenance difficulties later on. EditLog is used to handle persistent logs and cannot currently be used to pass information between FrontendServices. You can use FrontendService to pass information.

Reply:
fixed. changed to RPC